pkg/vsphere/adapter_test.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 VMware, Inc.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package vsphere
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"testing"
+	"time"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/binding"
+	"github.com/cloudevents/sdk-go/v2/client"
+	cecontext "github.com/cloudevents/sdk-go/v2/context"
+	"github.com/cloudevents/sdk-go/v2/event"
+	cehttp "github.com/cloudevents/sdk-go/v2/protocol/http"
+	"github.com/google/go-cmp/cmp"
+	"github.com/vmware/govmomi/vim25/types"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+type roundTripperTest struct {
+	statusCodes  []int
+	requestCount int
+	events       []*event.Event
+}
+
+func (r *roundTripperTest) RoundTrip(req *http.Request) (*http.Response, error) {
+	code := r.statusCodes[r.requestCount]
+	msg := cehttp.NewMessageFromHttpRequest(req)
+	e, err := binding.ToEvent(context.TODO(), msg)
+	if err != nil {
+		return nil, err
+	}
+	r.events = append(r.events, e)
+	return &http.Response{StatusCode: code}, nil
+}
+
+type mockType struct {
+	event *types.Event
+}
+
+func (m *mockType) GetEvent() *types.Event {
+	return m.event
+}
+
+func TestSendEvents(t *testing.T) {
+	now := time.Now()
+	baseEvent := &mockType{event: &types.Event{CreatedTime: now}}
+	testCases := map[string]struct {
+		statusCodes []int
+		moref       types.ManagedObjectReference
+		baseEvents  []types.BaseEvent
+		wantEvents  []*event.Event
+		result      error
+	}{
+		"one event, succeeds": {
+			statusCodes: []int{200},
+			moref:       types.ManagedObjectReference{Value: "VirtualMachine", Type: "vm59"},
+			baseEvents:  []types.BaseEvent{baseEvent},
+			wantEvents:  []*event.Event{createEvent("test-source", "mockType", "0", "eventex", nil, baseEvent, now)},
+		},
+		"one event, fails": {
+			statusCodes: []int{500},
+			moref:       types.ManagedObjectReference{Value: "VirtualMachine", Type: "vm59"},
+			baseEvents:  []types.BaseEvent{&mockType{event: &types.Event{CreatedTime: now}}},
```

Comment:
Could we reuse `baseEvent` from above?

Reply:
Absolutely, missed changing this when I added it abobve.